xbmc/addons/binary-addons/BinaryAddonManager.h

```diff
@@ -63,6 +63,24 @@ namespace ADDON
     bool HasEnabledAddons(const TYPE &type) const;
 
     /*!
+     * @brief Checks whether an addon is installed.
+     *
+     * @param[in] addonId id of the addon
+     * @param[in] type Add-on type to check installed
+     * @return true if installed
+     */
+    bool IsAddonInstalled(const std::string& addonId, const TYPE &type = ADDON_UNKNOWN);
+
+    /*!
+     * @brief Check whether an addon has been enabled.
+     *
+     * @param[in] addonId id of the addon
+     * @param[in] type Add-on type to check installed and enabled
+     * @return true if enabled
+     */
+    bool IsAddonEnabled(const std::string& addonId, const TYPE &type = ADDON_UNKNOWN);
+
```

Comment:
This is not something that should be added as public methods on BinaryAddonManager (same with GetDisabledAddonInfos). AddonMgr still have the authority over which addons are installed an enabled so all of these might give wrong/outdated answers.